A14-channelz.md

```diff
@@ -630,18 +646,48 @@ import "google/protobuf/any.proto";
 import "google/protobuf/duration.proto";
 import "google/protobuf/timestamp.proto";
 
-// Channel is a logical grouping of subchannels and sockets.
+// Channel is a logical grouping of channels, subchannels and sockets.
 message Channel {
   // The identifier for this channel.
   ChannelRef ref = 1;
   // Data specific to this channel.
   ChannelData data = 2;
-  ChannelTrace trace = 3;
-  // At most one of 'subchannel_ref' and 'socket' is set.
+  // At most one of 'channel_ref+subchannel_ref' and 'socket' is set.
```

Comment:
If you want to encode this restriction in the proto format, we could do something like this:

```
message ChannelAndSubchannelRefs {
  repeated ChannelRef channel_ref;
  repeated SubchannelRef subchannel_ref;
}
oneof {
  ChannelAndSubchannelRefs channel_and_subchannel_refs;
  repeated SocketRef socket_ref;
}
```

Reply:
I thought about this too, but it didn't seem significantly better.  It adds another level of indirection to getting at the data you want.  The amount of restriction here is already not enforceable the proto file, so I didn't want to partially do it here and partially in the comments.  

For example, can duplicate refs exist in the list?  What about refs with the same id, but different names?   Can they exist in both channel refs and sub channel refs?  Can channel 2 be a subchannel of channel 1, and an owned channel of channel 3?  


Lastly, there is another option that someone might reasonably ask for:

```
message ChannelAndSubchannelRef {
  oneof ref {
    ChannelRef channel_ref;
    SubchannelRef subchannel_ref;
  }
}
message ChannelAndSubchannelRefs {
  repeated ChannelAndSubchannelRef ref;
}
oneof {
  ChannelAndSubchannelRefs channel_and_subchannel_refs;
  repeated SocketRef socket_ref;
}
```

None of these seem like big enough problems to modify it.